File: frontend/src/types/library.ts
export enum LibraryType {
  MOVIE = 'movie',
  TV = 'tv',
}

export interface LibraryBase {
  name: string;
  root_path: string;
  type: LibraryType;
  description: string;
  scan_interval: number;
  auto_import: boolean;
  auto_import_scan_path?: string | null;
  auto_import_supported_formats?: string[] | null;
  activated_plugins?: Record<string, string[]>;
  is_public: boolean;
  is_active: boolean;
}

export interface LibraryCreate extends Omit<LibraryBase, 'root_path'> {
  user_id: string;
  root_path?: string | null;
}

export interface LibraryUpdate {
  name?: string | null;
  root_path?: string | null;
  description?: string | null;
  scan_interval?: number | null;
  auto_import?: boolean | null;
  auto_import_scan_path?: string | null;
  supported_formats?: string[] | null;
  activated_plugins?: Record<string, string[]> | null;
}

export interface LibraryInDB extends LibraryBase {
  id: string;
  user_id: string;
  is_deleted: boolean;
  deleted_at?: string | null;
  created_at: string;
  updated_at: string;
}

export type LibraryRead = LibraryInDB;

export interface LibraryPageResult {
  items: LibraryRead[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

/**
 * 路由层 Schema（请求/响应），与后端保持同名字段
 */
export interface LibraryCreateRequestSchema extends Omit<LibraryCreate, 'user_id'> {
  user_id?: string | null;
  root_path?: string | null;

  // 与后端 routers/schemas/library.py 路由层新增插件选择保持一致
  metadata_plugins?: string[] | null;
  subtitle_plugins?: string[] | null;
}

export type LibraryUpdateRequestSchema = LibraryUpdate;

export type LibraryReadResponseSchema = LibraryRead;

export type LibraryPageResultResponseSchema = LibraryPageResult;

File: backend/src/routers/schemas/library.py
from typing import List, Optional, Union
from pydantic import BaseModel, Field
from src.models import LibraryCreate, LibraryUpdate, LibraryPageResult, LibraryRead


class LibraryCreateRequestSchema(LibraryCreate):
    """创建库请求模型"""

    user_id: Optional[str] = Field(
        None, description="用户ID，由依赖注入完成，前端无需传递"
    )
    root_path: Optional[str] = Field(
        None, description="媒体库路径，系统根据其他字段自动拼接，无需传递"
    )
    # 新增：前端按类型选择插件（两个下拉列表）
    metadata_plugins: Optional[List[str]] = Field(
        None, description="启用的元数据类插件（如 tmdb/omdb）"
    )
    subtitle_plugins: Optional[List[str]] = Field(
        None, description="启用的字幕类插件（如 opensubtitles）"
    )


class LibraryUpdateRequestSchema(LibraryUpdate):
    """更新库请求模型"""

    pass  # 所有字段从 LibraryUpdate 继承


class LibraryReadResponseSchema(LibraryRead):
    """读取库响应模型"""

    pass  # 所有字段从 LibraryRead 继承，Pydantic 模型之间的兼容性机制 + FastAPI 的自动响应序列化机制，可直接 return LibraryRead


class LibraryPageResultResponseSchema(LibraryPageResult):
    """库分页结果响应模型"""

    pass
